adaptivefiltering/pdal.py

```diff
@@ -264,23 +298,100 @@ def save(self, filename, compress=False, overwrite=False):
+
+        If no segmentation is provided the user will be prompeted to draw one on a map of the dataset.
+
+
+        :param segmentation:
+        A Segmentation object, default: None
+        :type segmentation: Segmentation
+
+        :return:
+        DataSet
+
+        """
+
         if isinstance(segmentation, Segment):
             segmentation = Segmentation([segmentation])
 
-        # Construct an array of WKT Polygons for the clipping
-        polygons = [convert.geojson_to_wkt(s.polygon) for s in segmentation["features"]]
+        def apply_restriction(seg):
+
+            # raise an Error if two polygons are given.
+            if len(seg["features"]) > 1:
+                raise NotImplementedError(
+                    "The function to choose multiple segments at the same time is not implemented yet."
+                )
+            # Construct an array of WKT Polygons for the clipping
 
-        from adaptivefiltering.pdal import execute_pdal_pipeline
+            polygons = [convert.geojson_to_wkt(s["geometry"]) for s in seg["features"]]
+
+            from adaptivefiltering.pdal import execute_pdal_pipeline
+
+            # Apply the cropping filter with all polygons
+            newdata = execute_pdal_pipeline(
+                dataset=self, config={"type": "filters.crop", "polygon": polygons}
+            )
+
+            return PDALInMemoryDataSet(
+                pipeline=newdata,
+                provenance=self._provenance
+                + [
+                    f"Cropping data to only include polygons defined by:\n{str(polygons)}"
+                ],
+            )
+
+        # Maybe create the segmentation
+        if segmentation is None:
+            from adaptivefiltering.apps import create_segmentation
+
+            restricted = create_segmentation(self)
+            restricted._finalization_hook = apply_restriction
+
+            return restricted
+        else:
+            return apply_restriction(segmentation)
+
+    def convert_georef(self, spatial_ref_out="EPSG:4326", spatial_ref_in=None):
+        """Convert the dataset from one spatial reference into another.
+        :parma spatial_ref_out: The desired output format. The default is the same one as in the interactive map.
+        :type spatial_ref_out: string
+
+        :param spatial_ref_in: The input format from wich the conversation is starting. The faufalt is the last transformation output.
+        :type spatial_ref_in: string
+
+        """
+
+        # if no spatial reference input is given, iterate through the metadata and search for the spatial reference input.
+
+        if spatial_ref_in is None:
+
+            # spacial_ref_in = json.loads(self.pipeline.metadata)["metadata"]['readers.las']["comp_spatialreference"]
+            for keys, dictionary in json.loads(self.pipeline.metadata)[
+                "metadata"
+            ].items():
+                for sub_keys, sub_dictionary in dictionary.items():
```

Comment:
Isn't this loop equivalent to `spatial_ref_in = dictionary.get("comp_statialreference", None)`?